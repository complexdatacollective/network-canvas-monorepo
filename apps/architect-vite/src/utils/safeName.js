const safeName = (name) => name.replace(/[.$[\]{}]+/g, '');

export default safeName;
