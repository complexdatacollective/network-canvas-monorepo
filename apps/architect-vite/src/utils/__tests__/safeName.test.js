/* eslint-env jest */

import { range } from 'lodash';
import safeName from '../safeName';

const testString = range(32, 127).map((code) => String.fromCharCode(code)).join('');

describe('safeName()', () => {
  it('allows alphanumerics, dashes, and underscore', () => {
    expect(safeName(testString)).toMatchSnapshot();
  });
});
