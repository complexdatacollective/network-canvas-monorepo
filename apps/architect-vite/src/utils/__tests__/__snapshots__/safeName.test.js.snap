// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`safeName() allows alphanumerics, dashes, and underscore 1`] = `" !\\"#%&'()*+,-/0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ\\\\^_\`abcdefghijklmnopqrstuvwxyz|~"`;
